### PR TITLE
Added script for training resnet152 on multiple GPU's

### PR DESCRIPTION
**Pull request by Efrainq07**
The new script resnet152_DDP_training.py can be run on the command line with `python resnet152_DDP_training.py -g (number of gpus) -n (number of nodes)`. The default value for -n is 1 for one node. This is all done so resnet152 can be trained on multiple GPU's on multiple nodes so the time it takes for them to train can be measured.

### CODE REVIEW

File: examples/python/pytorch/resnet152_DDP_training.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import torch.optim as optim
+import argparse
+import torch.multiprocessing as mp
+from resnet_torch import resnet152
+import torchvision.transforms as transforms
+import time
+
+
+def setup(rank, world_size):
+    os.environ['MASTER_ADDR'] = 'localhost'
```
Comment (min-xu-ai) [inline examples/python/pytorch/resnet152_DDP_training.py:15]: this won't work with multiple nodes?

File: examples/python/pytorch/resnet152_DDP_training.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    os.environ['MASTER_PORT'] = '12355'
+    dist.init_process_group("nccl", rank=rank, world_size=world_size)
+
+def cleanup():
+    dist.destroy_process_group()
+
+def train(gpu, args):
```
Comment (min-xu-ai) [inline examples/python/pytorch/resnet152_DDP_training.py:22]: I would change arg "gpu" to "local_gpu_id" or even better "local_rank". We use local rank and global rank as standard term for a node's rank in a distributed training.

**Review by min-xu-ai — COMMENTED** [review]
LGTM. Some comments to make the code better.

File: examples/python/pytorch/resnet152_DDP_training.py (hunk 1 above)
Comment (Efrainq07) [reply examples/python/pytorch/resnet152_DDP_training.py:15]: Yeah, I would need to change the ip to the one from the rank 0 one right?

**Review by msbaines — APPROVED** [review]
LGTM.

**Review by jiazhihao — APPROVED** [review]

**Pull request merged**